api/services/Ban.js

```diff
@@ -87,19 +88,25 @@ exports.updateBanlist = async function (redToken, username, banlistEntry, friend
     console.log('Error: Could not find parsing marker in public banlist');
     throw {error: 'Error: Could not find parsing marker in public banlist'};
   }
-  var updated_content = '';
+  let updated_content;
   for (let i = start_index; i < end_index; i++) {
-    if (knownAlt && lines[i].match(new RegExp('/u/' + knownAlt)) || _.intersection(lines[i].match(/(\d{4}-){2}\d{4}/g), friend_codes).length) {
+    if (knownAlt && lines[i].includes('/u/' + knownAlt) || lines[i].includes('/u/' + username)
+        ||_.intersection(lines[i].match(/(\d{4}-){2}\d{4}/g), friend_codes).length) {
       // User was an alt account, modify the existing line instead of creating a new one
       let blocks = lines[i].split(' | ');
       if (blocks.length !== 4) {
         break;
       }
-      blocks[0] += ', /u/' + username;
+      blocks[0] = _.union(blocks[0].match(/\/?u\/[\w-]{1,20}/g), ['/u/' + username]).join(', ');
       blocks[1] = _.union(blocks[1].match(/(\d{4}-){2}\d{4}/g), friend_codes).join(', ');
-      blocks[3] = _.compact(_.union(blocks[3].split(', '), [igns])).join(', ');
+      try {
+        blocks[3] = Flairs.formatGames(Flairs.combineGames(Flairs.parseGames(blocks[3]), Flairs.parseGames(igns)));
```

Comment:
This prevents any duplicate entries from being created on the banlist. Now if two people ban someone at once, there won't be any side-effects (aside from the banned person getting two ban messages).